session.go

```diff
@@ -973,24 +973,70 @@ func (st *StmtTxn) SeekReverse(k kv.Key) (kv.Iterator, error) {
 	return kv.NewUnionIter(bufferIt, retrieverIt, true)
 }
 
-// StmtCommit implements the Transaction interface.
-func (st *StmtTxn) StmtCommit() error {
-	err := kv.WalkMemBuffer(st.buf, func(k kv.Key, v []byte) error {
+func (st *StmtTxn) cleanup() {
+	st.buf.Reset()
+	for key := range st.mutations {
```

Comment:
Can we just set st.mutations = nil?

Reply:
```
x.buf = nil
x.buf = make()
```

vs 

```
v.buf.reset()
```

Current code make less allocation...